projects/ui-components/lf-repository-browser/lf-repository-browser.component.ts

```diff
@@ -479,10 +486,19 @@ export class LfRepositoryBrowserComponent implements OnDestroy {
   }
 
   private async makeDataCall(parentEntry: LfTreeNode): Promise<ILfSelectable[]> {
-    this.lastDataCall = this.updateFolderChildrenAsync(parentEntry);
-    const selectable = await this.lastDataCall;
-    this.lastDataCall = undefined;
-    return selectable;
+    try {
+      this.hasError = false;
+      this.lastDataCall = this.updateFolderChildrenAsync(parentEntry);
+      const selectable = await this.lastDataCall;
+      this.lastDataCall = undefined;
+      return selectable;
+    }
+    catch (error) {
+      this.hasError = true;
+      this.isLoading = false;
+      this.ref.detectChanges();
+      throw error;
```

Comment:
why do you still need to throw the error?

Reply:
this was to not change the function signature to return undefined. Is there a another way around this?